datafusion/core/src/config.rs

```diff
@@ -288,26 +288,37 @@ impl ConfigOptions {
     }
 
     /// get a boolean configuration option
-    pub fn get_bool(&self, key: &str) -> bool {
+    pub fn get_bool(&self, key: &str) -> Option<bool> {
         match self.get(key) {
-            Some(ScalarValue::Boolean(Some(b))) => b,
-            _ => false,
+            Some(ScalarValue::Boolean(b)) => b,
+            Some(b) => Some(
+                b.to_string()
+                    .parse::<bool>()
+                    .unwrap_or_else(|_| panic!("Cannot parse bool from {:?}", &b)),
```

Comment:
🤔  I feel like in other PRs like https://github.com/apache/arrow-datafusion/issues/3316 are going in the opposite direction and removing the use of panic!

I would prefer this API returned None rather than panic'd (and maybe log a `warn!` message)

Reply:
I agree. We don't want to introduce more panics at this point. There is an effort to remove as many of the existing panics as is practical.